Add SDL_mutex.inc (v3.2.12)

suve opened the pull request.

suve committed:
Add SDL_mutex.inc (v3.2.12)

Free-Pascal-meets-SDL-Website commented on units/SDL_mutex.inc:
```diff
@@ -0,0 +1,746 @@
+ *
+ * \since This enum is available since SDL 3.2.0.
+ *}
+type
```
This enum should be translated according to our translation rules. (typed constants)

Free-Pascal-meets-SDL-Website commented on units/SDL_mutex.inc:
```diff
@@ -0,0 +1,746 @@
+ * \since This struct is available since SDL 3.2.0.
+ *}
+type
+  TSDL_InitState = record
```
This enum should be translated according to our translation rules. (typed constants)

suve commented:
Amended the commit to change the enum to typed constants. Also replaced `cbool` occurrences with `Boolean` and fixed one function missing a `cdecl` modifier.

Free-Pascal-meets-SDL-Website commented:
Great, thanks and merged!
 
(if we decide to change the Boolean name; we can update the Booleans in a separate PR)
